### PR TITLE
Experiment/bcmp time

### PR DESCRIPTION
**Pull request by victorsowa12**
Tried to include usage of new functions like node_id and bm_err_check. Lmk if I there are any other functions like those that I didn't use yet.

Here are some examples of setting and getting time from a bridge on "old" fw and a hello world mote on the "new" fw.

Bridge telling hello world mote to set its time:
Screenshot 2024-09-30 at 3 18 36 PM
Mote after having its time set from the bridge:
Screenshot 2024-09-30 at 3 18 46 PM
Mote getting the bridges time:
Screenshot 2024-09-30 at 3 19 09 PM
Bridge getting the motes time:
Screenshot 2024-09-30 at 4 28 36 PM
Mote setting the bridges time:
Screenshot 2024-09-30 at 4 29 59 PM
Bridge after mote set its time:
Screenshot 2024-09-30 at 4 29 51 PM

### CODE REVIEW

File: src/lib/bcmp/bcmp_time.cpp [hunk 1]
```diff
@@ -118,5 +168,31 @@ bool bcmp_time_process_time_message(bcmp_message_type_t bcmp_msg_type, uint8_t *
     }
   } while (0);
 
-  return should_forward;
+  if (should_forward) {
+    bcmp_ll_forward(data.header, data.payload, data.size, data.ingress_port);
```
Comment (matt001k) [inline src/lib/bcmp/bcmp_time.cpp]: Could we assign `err = bcmp_ll_forward(...)` here?

File: src/lib/bcmp/bcmp_time.cpp [hunk 2]
```diff
@@ -2,54 +2,89 @@
 #include "app_util.h"
 #include "bcmp.h"
 #include "bm_util.h"
-#include "device_info.h"
 #include "stm32_rtc.h"
```
Comment (matt001k) [inline src/lib/bcmp/bcmp_time.cpp:5]: Is there still a dependency on this?

Comment (victorsowa12) [reply src/lib/bcmp/bcmp_time.cpp:5]: Yeah I put some to-do's to abstract the RTC code away. That should probably be in bm_common right? something like a `bm_rtc.h` or `bm_time.h` file probably?

Comment (matt001k) [reply src/lib/bcmp/bcmp_time.cpp:5]: That sounds good! `bm_time` fits more I feel like.

**Review by matt001k — APPROVED** [review]

**Pull request merged**